app/src/sharedTest/java/org/oppia/android/app/recyclerview/BindableAdapterTest.kt

```diff
@@ -163,8 +163,10 @@ class BindableAdapterTest {
 
   @Test
   fun testSingleTypeAdapter_withOneViewType_noData_bindsNoViews() {
+    val testFragment = Fragment()
     // Set up the adapter to be used for this test.
-    TestModule.testAdapterFactory = { createSingleViewTypeNoDataBindingBindableAdapter() }
+    TestModule.testAdapterFactory =
```

Comment:
This mechanism can be simplified a bit, I think. We should update all of the ``create*Adapter(`` helpers to take in their respective factory, then we can update: https://github.com/oppia/oppia-android/blob/78f4f5255ded9c4bd64b582e0c50c5dd38a923e5/app/src/main/java/org/oppia/android/app/testing/BindableAdapterTestFragmentPresenter.kt#L35

to pass in both factories, then we can select whichever is needed & pass it to the corresponding helper when setting this factory, e.g.:

```kotlin
TestModule.testAdapterFactory = { singleFactory, _ -> createSingleViewTypeNoDataBindingBindableAdapter(singleFactory) }
```

This will ensure the correct fragment is hooked up since the test fragment will be injecting the factories directly. It also ensures that we're actually injecting the factories rather than constructing them like below.

Reply:
@BenHenning applied the refactor however classes which were testing lifecycle implementations have been affected and failing e.g testSingleTypeAdapter_setLifecycleOwnerTwice_throwsException. Please advice since we are not manually attaching lifecycle to the BindableAdapter and it is done through injection?

Reply:
Anything that is testing setLifecycleOwner behaviors can be removed. We should still verify that the lifecycle binding works, but since we can't change it now any tests that require setLifecycleOwner to be called after the initial binding, or not at all, can be removed as they're no longer possible behaviors.

Reply:
Okay Ben removing all tests covering LifecycleOwner